src/components/autocomplete/js/autocompleteController.js

```diff
@@ -638,14 +639,17 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     if (angular.isArray(items)) {
       handleResults(items);
     } else if (items) {
+      fetchesInProgress++;
       setLoading(true);
       promiseFetch = true;
       $mdUtil.nextTick(function () {
         if (items.success) items.success(handleResults);
         if (items.then)    items.then(handleResults);
         if (items.finally) items.finally(function () {
-          setLoading(false);
-          promiseFetch = false;
+          if (--fetchesInProgress === 0) {
+            setLoading(false);
+            promiseFetch = false;
```

Comment:
Yes definitely :+1: Just had no chance to look over, because of the rebased 52a519ead04f2faeba227fbc51a7214253cbf2ba commit.